clients/unshaded/src/main/java/tachyon/client/block/BufferedBlockInStream.java

```diff
@@ -29,13 +30,13 @@
 
 /**
  * Provides a stream API to read a block from Tachyon. An instance extending this class can be
- * obtained by calling {@link TachyonBlockStore#getInStream(long)}. The buffer size of the stream
- * can be set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a
- * block. This class is not thread safe and should only be used by one thread.
+ * obtained by calling {@link TachyonBlockStore#getInStream}. The buffer size of the stream can be
+ * set through configuration. Multiple {@link BufferedBlockInStream}s can be opened for a block.
  *
  * This class provides the same methods as a Java {@link InputStream} with additional methods from
  * Tachyon Stream interfaces.
  */
+@NotThreadSafe
```

Comment:
I agree with the benefit of having explicit `@NotThreadSafe`. Meanwhile, I am a little bit concerned about the verbosity of the code. My feeling (could be very biased or wrong) is there are classes we care  a lot about thread-safety (e.g., those classes representing an identity of service), and there are classes people typically care less (e.g., Options). Maybe we should have more discussion on this to see people's opinion.